controllers/tools.js

```diff
@@ -100,9 +109,27 @@ const runToolHandler = (req, res, next) => {
 	});
 };
 
+const runToolHandlerNestImplementation = (req, res, next) => {
+	Promise.all([
+		api(req, { version: 'v3' })
+			.get(`/tools?toolId=${req.params.ltiToolId}&courseId=${req.params.courseId}`),
```

Comment:
these parameters dont seem optional at all. I would suggest moving them into the route itself

Reply:
not ltitools, nevermind